storage_v2/factory.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package storage_v2
+
+import (
+	"context"
+)
+
+// Factory is a general factory interface to be reused across different storage factories.
+// It lives within the OTEL collector extension component's lifecycle.
+// The Initialize and Close functions supposed to be called from the
+// OTEL component's Start and Shutdown functions.
+type Factory interface {
```

Comment:
```suggestion
type FactoryBase interface {
```
Even though it is namespaced separately from the actual factories, this interface itself is not a "Factory"